compute/ARMComputeEx/src/core/NEON/kernels/NEOneHotKernel.cpp

```diff
@@ -90,6 +90,15 @@ Status validate_arguments(const ITensorInfo *indices, const ITensorInfo *depth,
 
   return Status{};
 }
+
+template <typename U, typename Enable = void> bool isOnValue(U) { return true; }
+
+template <typename U, typename std::enable_if<std::is_signed<U>::value>::type>
+bool isOnValue(U value)
+{
+  ARM_COMPUTE_ERROR_ON(value < -1);
```

Comment:
@ragmani It seems to throw error on `value < -1`. I think it should return `false`. 
Also, I think `indice >= depth` is valid input that we should not set to on_value.

Reply:
I misunderstood. I will fix it. Thanks for your pointing out.